### ISSUE TITLE
docname attribute not set for preview

### ISSUE THREAD

**Issue by pstueck** [asciidoctor/asciidoctor-intellij-plugin#475]
Hope that's the correct issue board &hellip; but since "asciidoctor-intellij-plugin" and "AsciidoctorFX" both having the same issue, chances are up, Asciidoctor itself could be the reason for the problem ;-)
Edit: bug crossposted to [AsciidocFX#474](https://github.com/asciidocfx/AsciidocFX/issues/474)

I am using {docname} to control some sections of my documents &hellip; but unfortunately the attribute is not set for previews neither in "asciidoctor-intellij-plugin" nor in "AsciidoctorFX".

When I have a document named "myADoc.adoc" with an active preview pane, a code snippet like &hellip;
```adoc
\{docname} is {docname}
\{outfilesuffix} is {outfilesuffix}
```
displays in the preview pane as
> {docname} is {docname}
{outfilesuffix} is .html

but what I would like to see is (same that is shown for **generated** HTML or PDF) &hellip;
> {docname} is myADoc
{outfilesuffix} is .html

Thus something like &hellip;
```adoc
ifeval::["myADoc" == "{docname}"]
// do some stuff specifically for this file
endif::[]
```
never does work in preview :-(

Regards
Peter

**Comment by ahus1** [comment]
@pstueck - thanks for reporting this issue. I'll look into this for the IntelliJ plugin. Please consider opening a second issue for the AsciidoctorFX editor.

**Comment by ahus1** [comment]
The next version of the plugin will set the following attributes for the preview: docname, docfilesuffix, docfile, docdir. 

They will also be part of auto-complete in the editor.

**Comment by ahus1** [comment]
This is now available in release 0.30.69 that I've published today.
